keras/utils/vis_utils.py

```diff
@@ -13,7 +13,10 @@
 
 
 def _check_pydot():
-    if not (pydot and pydot.find_graphviz()):
+    try:
+        # Attempt to create an image of a blank graph to check the pydot/graphviz installation.
+        pydot.Dot.create(pydot.Dot())
+    except:
```

Comment:
Oh wow... In this case let's catch the exception, add a comment on why we use `Exception`, and check that the exception message is what we except. `raise` otherwise.